### ISSUE TITLE
Seemly random plant loop solver failure with out-of-range fluid temperatures

### ISSUE THREAD

**Issue by jcyuan2020** [NREL/EnergyPlus#8752]
Issue overview
--------------
It was found that in a simulation case, the plant flow convergence is somehow affected by a few other settings which typically should not be sources of problems. Starting from the original file, it was reported that with auto sized plant flow rate, the simulation would encounter a flow solver problem due to out-of-range fluid temperature in April (in an annual simulation). However, by specifying a flow rate that is twice the auto-sized rate, the simulation would finish running without a problem. 

What’s later further found about the simulation is that in the original setting of auto-sizing, if you do dissect the simulation into two parts: one run with Jan 1 to March 30; and then another run from April 1 to Dec 31, the simulation would finish running both the sections without a problem. 

What’s even surprising is that in the original auto-sized case, if the running period is slightly changed from (e.g. from 2009 to 2010, plus the starting day change), the whole-year simulation would run through without any problems—no encountering with out-of-range flow temperatures. 

In a few other trials, the simulation could also occasionally occur in March time. So there seems to be no clear clue on what a deterministic condition that have triggered the problem. 

Some example error messages are like:
```
   ** Severe  ** Plant temperatures are getting far too hot, check controls and relative loads and capacities
…
   **   ~~~   ** PlantLoop Outlet Node (DemandSide) "OFCSML CWSYSTEM DEMAND MIXER OUTLET NODE" has temperature=1075.0 {C}
…
   **  Fatal  ** CheckForRunawayPlantTemps: Simulation terminated because of run away plant temperatures, too hot
```

It seems that there might be some rare (and probably random) conditions that would cause the plant solver to diverge. 

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version): All
 - Version of EnergyPlus (if using an intermediate build, include SHA): At least in 9.4 and 9.5
 - Helpdesk ticket number 15683

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added to EnergyPlusDevSupport\DefectFiles
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by nealkruis** [comment]
I started looking into this for the CBECC-Com team. Here's what I see happening:

When a WSHP operates at very low part load the calculated source-side energy is applied to a very small flow rate resulting in a very large outlet temperature. Users can get around this if they simply change the minimum design flow rate on the pump to be something like 1% of the design flow. This will force a minimum amount of water to go through the demand-side bypass branch and mix with that tiny amount of dangerously hot water coming out of the WSHP components.

There really should be something to put some realistic bounds on the condenser temperature where the water can't be heated above the temperature of the refrigerant.

**Comment by nealkruis** [comment]
I believe this has been resolved in #10043.
